### PR TITLE
chore: add technical guidelines

### PR DESCRIPTION
**Pull request by toddbaert**
* existing repo requirements and DCO section moved here from .github (will redirect them to here after merge)
* added semantic versioning + breaking changes section
* added platform support + removal section

These additions are based on the outcome of our TC meeting last week (cc @lukas-reining @thomaspoignant @kinyoklion). Python was used as a sort-of test case. Thanks @federicobond .

### CODE REVIEW

File: presentations.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ All items are listed in reverse chronological order from when they were first pr
...
 | ------------------------------------------------- | ---------------------- | ------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------- |
 | Feature Observability                             | KubeCon EU 24          | [Link](https://www.youtube.com/watch?v=euYhIn4leW0)                                                                 | [Link](https://docs.google.com/presentation/d/1ZvW-6dZdfJCH9qEMKAWPCp6aTzt2Us65)            |
-| Feature Management Improve                        | KubeCon EU 24          | [Link](https://www.youtube.com/watch?v=wkmryOXmVaw)                                                                 | [Link](https://docs.google.com/presentation/d/1aXp41SY7ChsYc728NXlo-vaX2Ekme1J5r9cmgwC5JYc) |
+| Feature Management Improv                         | KubeCon EU 24          | [Link](https://www.youtube.com/watch?v=wkmryOXmVaw)                                                                 | [Link](https://docs.google.com/presentation/d/1aXp41SY7ChsYc728NXlo-vaX2Ekme1J5r9cmgwC5JYc) |
```
Comment (toddbaert) [inline presentations.md:13]: unrelated

File: technical-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# Technical Guidelines
+
+## Developer Certificate Of Origin
```
Comment (toddbaert) [inline technical-guidelines.md:3]: This section is not new, taken for .github.

File: technical-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+```
+
+\* Branch protection rules should protect the primary branch (usually `main`) by requiring code review from the appropriate parties (other than the author), usually expressed in a CODEOWNERS file.
+
+\*\* We recommend Renovate over Dependabot because of it's group and auto-merge features.
+Additionally, we have an org-wide base config for Renovate.
+
+\*\*\* Release Please isn't strictly necessary, but combined with [Conventional Commits][conventional-commits], it provides a simple yet robust means of generating useful, accurate changelogs and releasing semantically versioned artifacts.
+
+## Repository requirements
```
Comment (toddbaert) [inline technical-guidelines.md:36]: This section is not new, taken for .github.

File: technical-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+| static analysis                                | [SonarCloud][sonarcloud], language-specific tools (SpotBugs, eslint)                                                                      | recommended                                                 |
+
+## Semantic Versioning and 1.0 Releases
+
+We require release artifacts to adhere to [semantic versioning](https://semver.org/) except in specific cases approved by the [Technical Steering Committee](https://github.com/open-feature/community/blob/main/community-members.md#technical-committee).
+1.0 releases must satisfy the [repository requirements](#repository-requirements) above.
+If the artifact is an SDK implementing the OpenFeature specification, it must also conform to a version of the OpenFeature specification not less than 2 minor versions behind the latest (ex: if the latest OpenFeature specification is `0.8.1`, then an implementation conforming to `0.6.0` is a candidate for `1.0` release, while an implementation conforming only to `0.5.0` is not).
```
Comment (toddbaert) [inline technical-guidelines.md:62]: This was a bit of a point of contention in the TC meeting, but I think a "moving target" is better than a static one (`0.6.0`). Open to further discussion here, of course.

**Review by thomaspoignant — APPROVED** [review]

File: technical-guidelines.md (hunk 4 above)
Comment (justinabrahms) [reply technical-guidelines.md:62]: There's a lot of importance being placed on 1.0 specifically, so it seems worth noting that while OF spec is 0.x, minor releases can have backwards incompatible changes. Releasing 1.0 may necessitate 2.0 for sdk authors.

File: technical-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+Please consult with a member of the [TC](https://github.com/open-feature/community/blob/main/community-members.md#technical-committee) before making breaking changes to a `1.0+` component.
+
+## Platform Support
+
+OpenFeature maintains SDKs and integrations targeting various platforms, runtimes, and frameworks.
+This section describes our approach to supporting, and deprecating support for these.
+
+### Platform Support Recommendations
+
+* Efforts should be made to support the usage of OpenFeature libraries on all officially supported (non-EOL) platforms, at a minimum.
```
Comment (justinabrahms) [inline technical-guidelines.md:87]: can we link "all officially supported platforms" here to wherever we define that?

**Review by justinabrahms — APPROVED** [review]

File: technical-guidelines.md (hunk 5 above)
Comment (toddbaert) [reply technical-guidelines.md:87]: I had trouble wording this line. What I meant was all supported versions of the platform the SDK targets, ie: for node, the current supported versions are [18/20](https://endoflife.date/nodejs). The idea is if you are writing a C# SDK, you should support the currently officially supported versions of C#, etc.

Comment (weyert) [reply technical-guidelines.md:87]: guess, you could say officially supported/maintained versions up to two versions or something?

File: technical-guidelines.md (hunk 4 above)
Comment (beeme1mr) [reply technical-guidelines.md:62]: Wouldn't it be reasonable for a new SDK to be fully spec complient? We don't update the spec that often.

File: technical-guidelines.md (hunk 5 above)
Comment (kinyoklion) [reply technical-guidelines.md:87]: I think we may better need to define what a platform is in respect to an SDK. We are talking about language, frameworks, and virtual machines. Maybe sometimes we will be talking about OS as well (iOS, Android, C++). Go for Go we are talking about the Go version, not Windows 11, for instance. For some SDKs you my need 1 more level. The react SDK would likely have version constraints on react, browsers, and maybe node.

Comment (kinyoklion) [reply technical-guidelines.md:87]: Maybe we can provide some examples and then defer, to some extent, what a platform is to the SDK.

The platforms being targeted also evolve and that may provide some leeway. You may make a server SDK and it supports node. Eventually it supports deno, bun, edge environments, and incrementally has more platform requirements.

File: technical-guidelines.md (hunk 4 above)
Comment (maxveldink) [reply technical-guidelines.md:62]: Do we want to make any differentiation between sections of the spec? Is an experimental section still required for a 1.0 release? Or only once it's promoted to hardening?

File: technical-guidelines.md (hunk 5 above)
Comment (lukas-reining) [reply technical-guidelines.md:87]: I was also thinking examples for `runtimes`, `language versions` or `OS`s` could be helping this point.

**Review by lukas-reining — APPROVED** [review]
Looks good! :)

File: technical-guidelines.md (hunk 5 above)
Comment (toddbaert) [reply technical-guidelines.md:87]: I've updated this to `Efforts should be made to support the usage of OpenFeature libraries on all officially supported (non-EOL) platforms for the library in question (i.e.: a library for Node.js should support all officially supported versions of Node.js).`

File: technical-guidelines.md (hunk 4 above)
Comment (toddbaert) [reply technical-guidelines.md:62]: This is a good point. I've tried to address this with additional changes:

```markdown
> [!IMPORTANT]  
> While the OpenFeature specification version is <`1.0`, it's possible breaking changes will be introduced.
> This may necessitate that SDKs which have released `1.0` versions, release `2.0` versions in order to adhere to breaking specification changes.
```

File: technical-guidelines.md (hunk 5 above)
Comment (toddbaert) [reply technical-guidelines.md]: > Maybe we can provide some examples and then defer, to some extent, what a platform is to the SDK.

> The platforms being targeted also evolve and that may provide some leeway. You may make a server SDK and it supports node. Eventually it supports deno, bun, edge environments, and incrementally has more platform requirements.

Good points.

Comment (toddbaert) [reply technical-guidelines.md]: @kinyoklion To the preceding section, I've added:

```
For the purposes of this document, a "platform" constitutes an assumed underlying runtime dependency.
This includes, but is not limited to: languages, operating systems, dependency managers, virtual machines and software libraries.
```

**Review by beeme1mr — APPROVED** [review]
